pkg/controller/sparkapplication/controller.go

```diff
@@ -296,6 +298,14 @@ func (c *Controller) getAndUpdateDriverState(app *v1beta1.SparkApplication) erro
 			if pod.Status.Phase == apiv1.PodSucceeded || pod.Status.Phase == apiv1.PodFailed {
 				currentDriverState.completionTime = metav1.Now()
 			}
+			// Fetch container ExitCode/Reason if Pod Failed.
+			if pod.Status.Phase == apiv1.PodFailed {
+				if len(pod.Status.ContainerStatuses) > 0 && pod.Status.ContainerStatuses[0].State.Terminated != nil {
```

Comment:
Looks like it worths creating a variable for `pod.Status.ContainerStatuses[0].State` as its repeated a few times here.

Reply:
Created one for pod.Status.ContainerStatuses[0].State.Terminated  after validating  that the length of ContainerStatuses ?